### DISCUSSION TITLE
ksh 93u+m/1.0.0-rc.1

### DISCUSSION THREAD

**Discussion by McDutchie** [ksh93/ksh#492] [General]
We're nearly there!

I intend to release ksh 93u+m/1.0.0 on 2022-08-01, precisely ten years after the last canonical 93u+ release.

We have a week until then, so here's a release candidate. Please try as hard as you can to break it, and to help fix known bugs. You can report your findings or just talk about it on the GitHub discussion thread associated with this release.

**Full Changelog**: https://github.com/ksh93/ksh/compare/v1.0.0-beta.2...v1.0.0-rc.1

<hr /><em>This discussion was created from the release <a href='https://github.com/ksh93/ksh/releases/tag/v1.0.0-rc.1'>ksh 93u+m/1.0.0-rc.1</a>.</em>

**Comment by McDutchie** [comment]
One thing to consider before release: upgrade the licence to [EPL 2.0](https://www.eclipse.org/legal/epl-2.0/faq.php), which is explicitly allowed by EPL 1.0 in [section 7](https://github.com/ksh93/ksh/blob/dev/LICENSE.md#7-general), second to last paragraph. See [this commit in Lev Kujawski's AST fork](https://github.com/lkujaw/ast/commit/b2a8b30d1c2c9de3465d388f4f410169a3cb4df5) for details and a rationale for why he upgraded it in his fork. I'm not a lawyer but it does seem like it is okay to do this. I don't have strong feelings about it though. Does anyone else have an opinion one way or another?

**Reply by johnsonjh** [reply]
Hi - I like EPL2 > 1.  If for no other reason than the Eclipse foundation has recommended 2 > 1 and deprecated V1.  There is extensive documentation and details on V2 (https://www.eclipse.org/legal/epl-2.0/faq.php) which is much more than is available for the earlier license.  

I've looked over both and see no downsides to the new license, and the new terminology is clearer (files vs. modules, etc.)

**Reply by McDutchie** [reply]
Alright, might as well upgrade the licence. ~~Working on it.~~ Done.

**Comment by hyenias** [comment]
Here are my tests results using the most current 1.0 release candidate https://github.com/ksh93/ksh/commit/b07d7b209798dbe6048e2a1aa4ccdd5695ddb3e2:
```
$ arch/*/bin/ksh --version
  version         sh (AT&T Research) 93u+m/1.0.0-rc.5 2022-07-31

freebsd13.1 arm64-64: all tests pass

debian11 aarch64: all tests pass

ubuntu20.04 armv7l: tilde(shcomp) fails ~1 out of 3 attempts
$ bin/shtests tilde
#### Regression-testing /dev/shm/ksh/arch/linux.arm/bin/ksh ####
test tilde begins at 2022-07-31+15:34:10
test tilde passed at 2022-07-31+15:34:10 [ 24 tests 0 errors ]
test tilde(C.UTF-8) begins at 2022-07-31+15:34:10
test tilde(C.UTF-8) passed at 2022-07-31+15:34:10 [ 24 tests 0 errors ]
test tilde(shcomp) begins at 2022-07-31+15:34:10
	shcomp-tilde.ksh[155]: FAIL: get discipline: quoting of whitespace: expected $'3,one\ttwo  three\n\tfour,one\ttwo  three\n\tfour,one\ttwo  three\n\tfour', got $'3,one\ttwo,one\ttwo  three\n\tfour,one\ttwo  three\n\tfour'
test tilde(shcomp) failed at 2022-07-31+15:34:10 with exit code 1 [ 24 tests 1 error ]

ubuntu18.04 x86_64: rare failure on path.sh:
test path begins at 2022-07-31+16:02:58
test path passed at 2022-07-31+16:03:03 [ 146 tests 0 errors ]
test path(C.UTF-8) begins at 2022-07-31+16:03:03
	path.sh[569]: FAIL: 'command -x' test yielded exit status 267/SIGSEGV
	path.sh[577]: FAIL: 'command -x' did not correctly divide arguments (expected 802813 args of total size 9983542, got 175802 args of total size 2187441; divided in 2 chunks)
test path(C.UTF-8) failed at 2022-07-31+16:03:05 with exit code 2 [ 146 tests 2 errors ]
test path(shcomp) begins at 2022-07-31+16:03:05
test path(shcomp) passed at 2022-07-31+16:03:08 [ 146 tests 0 errors ]

openbsd7.1 amd64: intermittent failure on io.sh and failures on variables.sh :
test io(C.UTF-8) begins at 2022-07-31+14:40:30
	io.sh[348]: FAIL: read -n3 from fifo failed -- expected 'a', got 'abc'
	io.sh[351]: FAIL: read -n1 from fifo failed -- expected 'b', got 'd'
test io(C.UTF-8) failed at 2022-07-31+14:40:37 with exit code 2 [ 162 tests 2 errors ]
test variables begins at 2022-07-31+14:48:32
/mnt/tmp/ksh/src/cmd/ksh93/tests/variables.sh: line 849: x.b: no parent
test variables failed at 2022-07-31+14:48:39 with exit code 1 [ 192 tests 1 error ]
test variables(C.UTF-8) begins at 2022-07-31+14:48:39
/mnt/tmp/ksh/src/cmd/ksh93/tests/variables.sh: line 849: x.b: no parent
test variables(C.UTF-8) failed at 2022-07-31+14:48:46 with exit code 1 [ 192 tests 1 error ]
test variables(shcomp) begins at 2022-07-31+14:48:46
/tmp/ksh93.shtests.11792.32127/shcomp-variables.ksh: line 849: x.b: no parent
test variables(shcomp) failed at 2022-07-31+14:48:53 with exit code 1 [ 192 tests 1 error ]
```

**Reply by McDutchie** [reply]
Thanks for testing, @hyenias. I'm not close to figuring out the cause of any of those intermittent failures, hopefully they can be fixed in a future release.
